File: backend/src/lib/validate.js
'use strict';

function requireFields(body, fields) {
  const missing = fields.filter(f => body[f] === undefined || body[f] === null || body[f] === '');
  if (missing.length) throw new Error(`Campos requeridos faltantes: ${missing.join(', ')}`);
}

function requireMinProps(obj, min, name = 'properties') {
  if (!obj || typeof obj !== 'object' || Object.keys(obj).length < min) {
    throw new Error(`${name} debe tener al menos ${min} propiedades`);
  }
}

module.exports = { requireFields, requireMinProps };

File: backend/apps/core/lib/validate.py
"""Validaciones simples sobre payloads."""
from rest_framework.exceptions import ValidationError


def require_fields(data, fields):
    missing = [
        f for f in fields
        if data.get(f) is None or data.get(f) == ''
    ]
    if missing:
        raise ValidationError({'detail': f"Campos requeridos faltantes: {', '.join(missing)}"})


def require_min_props(obj, minimum, name='properties'):
    if not isinstance(obj, dict) or len(obj) < minimum:
        raise ValidationError({'detail': f"{name} debe tener al menos {minimum} propiedades"})
